hibernate-core/src/main/java/org/hibernate/action/internal/EntityAction.java

```diff
@@ -101,19 +100,7 @@ public String getEntityName() {
 	 */
 	public final Serializable getId() {
 		if ( id instanceof DelayedPostInsertIdentifier ) {
-			final EntityEntry entry = session.getPersistenceContext().getEntry( instance );
-			if ( entry == null ) {
-				if ( LOG.isDebugEnabled() ) {
-					LOG.debugf(
-							"Skipping action - the persistence context does not contain any entry for the entity [%s]. This may occur if an entity is created and then deleted in the same transaction/flush.",
-							instance
-					);
-				}
-				// If an Entity is created and then deleted in the same Transaction, when Action#postDelete() calls this method the persistence context
-				// does not contain anymore an entry.
-				return null;
-			}
-			final Serializable eeId = entry.getId();
+			final Serializable eeId = session.getPersistenceContext().getEntry( instance ).getId();
 			return eeId instanceof DelayedPostInsertIdentifier ? null : eeId;
```

Comment:
Is it possible for `session.getPersistenceContext().getEntry( instance )` to be `null`? If so, NPE will be thrown.

Reply:
Since it's technically possible, I'll revert back to the null check for safety.